[custom_op] Add a test for symints

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #101824
* #101823
* __->__ #101822

Tests that a custom op annotated with Sequence[int] actually accepts
Sequence[SymInt].

zou3519 committed:
[custom_op] Add a test for symints

Tests that a custom op annotated with Sequence[int] actually accepts
Sequence[SymInt].

[ghstack-poisoned]

ezyang reviewed (approved):
You might consider doing an assertExpectedInline on the traced graph

albanD commented on torch/testing/_internal/custom_op_db.py:
```diff
@@ -126,6 +126,22 @@ def sample_inputs_numpy_nonzero(opinfo, device, dtype, requires_grad, **kwargs):
+
+@numpy_view_copy.impl(['cpu', 'cuda'])
+def numpy_view_copy_impl(x, shape) -> Tensor:
+    return torch.tensor(np.copy(to_numpy(x).reshape(shape)), device=x.device)
```
Why not `torch.from_numpy()` ?

albanD reviewed (approved):
Small nit only

zou3519 committed:
Update on "[custom_op] Add a test for symints"


Tests that a custom op annotated with Sequence[int] actually accepts
Sequence[SymInt].

[ghstack-poisoned]

zou3519 committed:
Update on "[custom_op] Add a test for symints"


Tests that a custom op annotated with Sequence[int] actually accepts
Sequence[SymInt].

[ghstack-poisoned]
